Alikins/dnf rebaseier

alikins opened the pull request.

alikins commented:
This is https://github.com/candlepin/subscription-manager/pull/1190 plus rebase plus spec file support, minor bug fixes, and style things.

dgoodwin commented:
**Build Log**
*last 10 lines*

 ```
[...truncated 958 lines...]
[htmlpublisher] Archiving at BUILD level /home/jenkins/workspace/subscription-manager-nose-tests-pull-request-builder/cover to /jenkins-results/rhsm-jenkins/jobs/subscription-manager-nose-tests-pull-request-builder/builds/374/htmlreports/Coverage_module_html_report
Recording test results
Sending e-mails to: chainsaw@redhat.com
IRC notifier plugin: Sending notification to: #candlepin
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Setting status of 316f150681425dd019f49439cedb5f6090f7d80c to FAILURE with url http://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/subscription-manager-nose-tests-pull-request-builder/374/ and message: 'Build finished.'
```

dgoodwin commented:
**Build Log**
*last 10 lines*

 ```
[...truncated 958 lines...]
[htmlpublisher] Archiving at BUILD level /home/jenkins/workspace/subscription-manager-nose-tests-pull-request-builder/cover to /jenkins-results/rhsm-jenkins/jobs/subscription-manager-nose-tests-pull-request-builder/builds/379/htmlreports/Coverage_module_html_report
Recording test results
Sending e-mails to: chainsaw@redhat.com
IRC notifier plugin: Sending notification to: #candlepin
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Setting status of 9a30e76136a9842653f710201efbda8c64a9ba1b to FAILURE with url http://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/subscription-manager-nose-tests-pull-request-builder/379/ and message: 'Build finished.'
```

wottop commented:
retest this please

dgoodwin commented:
**Build Log**
*last 10 lines*

 ```
[...truncated 957 lines...]
[htmlpublisher] Archiving at BUILD level /home/jenkins/workspace/subscription-manager-nose-tests-pull-request-builder/cover to /jenkins-results/rhsm-jenkins/jobs/subscription-manager-nose-tests-pull-request-builder/builds/422/htmlreports/Coverage_module_html_report
Recording test results
Sending e-mails to: chainsaw@redhat.com
IRC notifier plugin: Sending notification to: #candlepin
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Setting status of 9a30e76136a9842653f710201efbda8c64a9ba1b to FAILURE with url http://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/subscription-manager-nose-tests-pull-request-builder/422/ and message: 'Build finished. 1736 tests run, 0 skipped, 18 failed.'
```

alikins commented:
Rebased this and fixed some of the tests. I think this in combo with alikins/repolib_split will let us move yum/dnf related code into a sub package, so systems without either can skip them.

cnsnyder commented on test/test_productid.py:
```diff
@@ -9,6 +9,7 @@
 import stubs
 from subscription_manager import productid
 from subscription_manager import certdirectory
+#from subscription_manager.plugin.yum impor
```
Removable?

alikins replied:
yup. I'll push a clean up (and likely a rebase/squash)

cnsnyder commented on test/test_productid.py:
```diff
@@ -583,25 +488,17 @@ def test_update_installed_no_active_with_product_certs_installed_anaconda(self):
         mock_yb = Mock(spec=yum.YumBase)
         mock_yb.pkgSack.returnPackages.return_value = []
-        mock_yb.repos.listEnabled.return_value = []
+        #mock_yb.repos.listEnabled.return_value = []
```
Do we need this as a comment?

alikins commented:
Cleaned up commented out code and squashed some commits and force pushed.

cnsnyder commented:
Looks good, ACK
